integration/fuse/src/main/java/alluxio/fuse/AlluxioFuse.java

```diff
@@ -194,9 +194,10 @@ public static FuseUmountable launchFuse(FileSystemContext fsContext, FileSystem
         Signal.handle(new Signal("TERM"), fuseSignalHandler);
 
         try {
+          String[] fuseOptsArray = fuseOpts.toArray(new String[fuseOpts.size()]);
```

Comment:
It's ok to use a zero-length array. But the difference between the zero-length array and non-zero length array is that if the length of the array is less than the collection's size, a new array of the required length and the same type is allocated. Otherwise the array will be filled and return.
ref [Collection.toArray()](https://www.baeldung.com/java-collection-toarray-methods)

Reply:
https://stackoverflow.com/questions/174093/toarraynew-myclass0-or-toarraynew-myclassmylist-size
The recommendation for at least Java 8 is to use zero sized array. There might be some internal JVM caveats that makes that faster. Also I believe Intellij will suggest you to use the zero sized array?

Reply:
Surprising trick! Thanks @jiacheliu3 @dbw9580